content/blog/design-kgateway-for-scalability.md

```diff
@@ -0,0 +1,170 @@
+---
+title: Design kgateway for Scalability – Not All Gateways Are Created Equal
+toc: false
+publishDate: 2025-05-07T00:00:00-00:00
+author: Lin Sun & Yuval Kohavi
+---
+
+With the Kubernetes Gateway API becoming the de facto standard for managing traffic into, out of, and within clusters, a growing number of gateways now implement this API. While gateways are often expected to "just work," choosing the right one can have major implications—especially at scale.
+
+Before comparing features, the most critical consideration is whether a gateway is built on a solid, reliable foundation–the kind of foundation that just works.
+
+## Aren’t All Envoy-Based Gateways the Same?
+
+It's a fair question. Many gateways use the Envoy proxy under the hood to enforce routing, security, and other traffic policies. But despite sharing the same data plane, not all gateways are equal—because the **control plane** makes all the difference.
+
+The control plane is responsible for translating Kubernetes Gateway API resources into actual Envoy configuration. This translation layer can be simple for a few routes, but when you’re managing 20,000 resources (which will translate into 500,000+ lines of Envoy config), the efficiency and scalability of the control plane become critical—as I noted in a previous [LinkedIn post](https://www.linkedin.com/posts/lin-sun-a9b7a81_from-20000-kubernetes-gateway-resources-activity-7305695363049373696-UVNz?utm_source=share&utm_medium=member_desktop&rcm=ACoAAABLihcBuozqLyftNtauegAdN2-QszsmqQQ).
```

Comment:
I would remove the LinkedIn post reference, it doesn't add anything new

Reply:
i like keep it in to keep history as that post has a lot of impressions.